d3a_interface/kafka_communication/kafka_producer.py

```diff
@@ -0,0 +1,60 @@
+import logging
+import json
+from zlib import compress
+from kafka import KafkaProducer
+
+from d3a_interface.kafka_communication import KAFKA_URL, DEFAULT_KAFKA_URL, KAFKA_USERNAME, \
+    KAFKA_PASSWORD, KAFKA_COMMUNICATION_SECURITY_PROTOCOL, KAFKA_SASL_AUTH_MECHANISM, \
+    KAFKA_API_VERSION, create_kafka_new_ssl_context, KAFKA_TOPIC
+
+KAFKA_PUBLISH_RETRIES = 5
+KAFKA_BUFFER_MEMORY = 2048000000
+KAFKA_MAX_REQUEST_SIZE = 2048000000
+
+
+def kafka_connection_factory():
+    try:
+        return KafkaConnection()
+    except Exception:
+        logging.info("Running without Kafka connection for simulation results.")
+        return DisabledKafkaConnection()
+
+
+class DisabledKafkaConnection:
+    def __init__(self):
+        pass
+
+    def publish(self, endpoint_buffer):
+        pass
+
+    @staticmethod
+    def is_enabled():
+        return False
+
+
+class KafkaConnection:
```

Comment:
To make sure we have the same interfaces here do you think it's okay if we extend the `DisabledKafkaConnection` class here ?

Reply:
Well by only inheriting from the class we do not ensure that we have the same interfaces, however I understand your point and I agree that it can be achieved by converting the DisabledKafkaConnection to an abstract base class. Let me try that, thanks!

Reply:
I unfortunately had to revert this fix because I forgot that one cannot instantiate an ABC, there needs to be a third class that inherits from the ABC. In order to not have 3 classes I think it is simpler to not define an ABC at all, at least until the interface becomes more complex that justifies the extra class for the interface.